### PR TITLE
maitake: change name to `maitake`, add docs

### PR DESCRIPTION
**Pull request by hawkw**

### CODE REVIEW

File: maitake/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+[sponsor-url]: https://github.com/sponsors/hawkw
+[maitake-wiki]: https://en.wikipedia.org/wiki/Grifola_frondosa
+
+## what is it?
+
+This library is a collection of modular components for building a Rust
+async runtime based on [`core::task`] and [`core::future`], with a focus on
+supporting `#![no_std]` projects.
+
+Unlike other async runtime implementations, `maitake` does *not* provide a
```
Comment (jamesmunns) [inline maitake/README.md:35]: Compare to `smol` perhaps?

File: maitake/README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+`maitake` was initially designed for use in the [mycelium] and [mnemOS]
+operating systems, but may be useful for other projects as well.
+
+[`core::task`]: https://doc.rust-lang.org/stable/core/task/index.html
+[`core::future`]: https://doc.rust-lang.org/stable/core/future/index.html
+[task system]: task
+[scheduling]: scheduler
+[wait]: wait
+[mycelium]: https://github.com/hawkw/mycelium
+[mnemOS]: https://github.com/jamesmunns/pellegrino/tree/main/firmware/kernel
```
Comment (jamesmunns) [inline maitake/README.md:51]:
```suggestion
[mnemOS]: https://mnemos.jamesmunns.com
```

File: maitake/README.md (hunk 1 above)
Comment (hawkw) [reply maitake/README.md:35]: oh good point, smol is in _some_ ways similar, but the primitives it exposes are somewhat higher-level

Comment (jamesmunns) [reply maitake/README.md:35]: Gotcha, you know better than me! I just knew that `smol` had been re-used in multiple executors as sort of a "core" building block.

Comment (hawkw) [reply maitake/README.md:35]: i might add a whole "comparison to other crates" section in a follow-up PR?

**Review by jamesmunns — APPROVED** [review]
Overall looks good to me! Made two small suggestions, feel free to merge when you're happy.

:partying_face:

File: maitake/README.md (hunk 1 above)
Comment (jamesmunns) [reply maitake/README.md:35]: Sounds good! The natural comparisons would be embassy (also no_std focused, but different), smol (as another building-block-ish crate), and tokio/async-std (as full-featured, but std-focused runtimes/ecosystems)

Comment (hawkw) [reply maitake/README.md:35]: > Gotcha, you know better than me! I just knew that `smol` had been re-used in multiple executors as sort of a "core" building block.

yeah, it's still a slightly higher level primitive, i think --- this crate is more comparable to something like [`async-task`](https://docs.rs/async-task/latest/async_task/) which is a dependency of `smol`.

Comment (jamesmunns) [reply maitake/README.md:35]: Specifically for embassy, maybe focus on the fact that maitake is optionally-alloc, and supports dynamic task spawning

**Pull request merged**